src/pyqasm/transformer.py

```diff
@@ -225,15 +226,24 @@ def transform_gate_params(
             gate_op.argument = Qasm3Transformer.transform_expression(gate_op.argument, param_map)
 
     @staticmethod
-    def get_branch_params(condition: Any) -> tuple[Optional[int], str, Optional[bool]]:
+    def get_branch_params(
+        condition: Expression,
+    ) -> tuple[
```

Comment:
It is not immediately evident what this function returns until we gloss through the docs. I think defining a `namedtuple` would make the definition clearer, something like - 
```python
from collections import namedtuple 
BranchParams = namedtuple("BranchParams", ["reg_id", "reg_name", "op", "rhs_val"])
```
This `BranchParams` can be used as the return type of the `get_branch_params` function. If the values are getting updated in visitor, feel free to used `dataclass`.